File: parser/src/main/java/io/ticofab/androidgpxparser/parser/domain/TrackPoint.java
package io.ticofab.androidgpxparser.parser.domain;

/**
 * A track point (trkpt) element.
 */
public class TrackPoint extends Point {

    private TrackPoint(Builder builder) {
        super(builder);
    }

    public static class Builder extends Point.Builder {

        @Override
        public TrackPoint build() {
            return new TrackPoint(this);
        }
    }

}

File: parser/src/main/kotlin/io/ticofab/androidgpxparser/parser/domain/TrackPoint.kt
package io.ticofab.androidgpxparser.parser.domain

/**
 * A track point (trkpt) element.
 */
class TrackPoint private constructor(builder: Builder) : Point(builder) {
    class Builder : Point.Builder() {
        override fun build(): TrackPoint {
            return TrackPoint(this)
        }
    }
}